### PR TITLE
Create codeql.yml

### LINKED ISSUE #147: Replace travis with GH actions

### PR DESCRIPTION
**Pull request by laDok8**
Move from travis to GHA
resolves #147

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (istraka) [inline .github/workflows/codeql.yml:16]: The main was changed from "master" to "main"

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  pull_request:
```
Comment (istraka) [inline .github/workflows/codeql.yml:17]: I am not sure if we want to run the static analysis on every PR. Can you provide a link to a test run of the workflow? What if there is a false-positive? How can I define an exception?

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '0 0 * * 0'
```
Comment (istraka) [inline .github/workflows/codeql.yml:21]: If nothing is merged, there is no need to run this once per week.

**Review by istraka — CHANGES_REQUESTED** [review]
Thanks a lot @laDok8!

I have two small requests and one question. See the inline comments.

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (laDok8) [reply .github/workflows/codeql.yml:17]: PR can be merged even with a check failing, but It's not recommended and should be reported eg. 
 https://github.com/github/codeql/issues/10766

here is an example run:
https://github.com/laDok8/sunstone/actions/runs/4165030065
vulnerabilities results are shown in the security tab of the repository, but they are private so I cannot link them

**Comment by laDok8** [comment]
Few things to note regarding codeQL:

- by default, no queries are run and must be uncommented in yml file
- check/job fails only if code cannot be compiled
- results are shown in the security tab of the repository and aren't publicly visible
- alerts can be marked as won't fix/ false positive and thus are not tracked anymore ( even with line change)
- alerts can be closed with a comment but no discussion can be led on them
- alerts can be reopened at any time
- when a faulty alert is moved to other file/line old alert is closed and a new one is opened without reference
- alerts show useful info eg.: affected branches, commit when the bug was introduced, history ...

**Comment by istraka** [comment]
Thanks Lada!

**Review by istraka — APPROVED** [review]

**Pull request merged**